src/app/containers/PageHandlers/withHashChangeHandler/index.test.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import withHashChangeHandler from '.';
+
+const Fixture = withHashChangeHandler(() => (
+  <>
+    <a href="#section-1">Go to section 1</a>
+    <section id="section-1">Section 1</section>
+  </>
+));
+
+window.HTMLElement.prototype.scrollIntoView = jest.fn();
+window.HTMLElement.prototype.focus = jest.fn();
+
+it('should scroll into view and focus on element when hash location changes', async () => {
+  const { rerender } = render(<Fixture location={{ hash: '' }} />);
+
+  rerender(<Fixture location={{ hash: '#section-1' }} />);
+
+  expect(window.HTMLElement.prototype.scrollIntoView).toHaveBeenCalledTimes(1);
```

Comment:
not a great unit test tbh as it's just testing implementation details.

Reply:
Is this more the sort of thing you are after, checking the active element? https://github.com/bbc/simorgh/blob/093faa31ea968cc2d1e68ce681dd27c3d9554498/src/app/containers/ConsentBanner/a11y.canonical.test.jsx#L53

Reply:
`document.activeElement` won't work in this case with JSDOM unless I use react testing library's `fireEvent` to send a click event.